component/org.wso2.carbon.identity.oauth2.token.handler.clientauth.mutualtls/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/mutualtls/internal/MutualTLSServiceComponent.java

```diff
@@ -37,23 +41,27 @@
 public class MutualTLSServiceComponent {
 
     private static final Log log = LogFactory.getLog(MutualTLSServiceComponent.class);
-    private BundleContext bundleContext;
 
     @Activate
     protected void activate(ComponentContext context) {
 
         try {
-            // Registering MutualTLSClientAuthenticator as an OSGIService.
-            bundleContext = context.getBundleContext();
+            BundleContext bundleContext = context.getBundleContext();
             MutualTLSClientAuthenticator mutualTLSClientAuthenticator = new MutualTLSClientAuthenticator();
+            IntrospectionResponseInterceptor introspectionResponseInterceptor = new IntrospectionResponseInterceptor();
+            ISIntrospectionDataProvider isIntrospectionDataProvider = new ISIntrospectionDataProvider();
             bundleContext.registerService(OAuthClientAuthenticator.class.getName(), mutualTLSClientAuthenticator,
                     null);
+            bundleContext.registerService(OAuthEventInterceptor.class.getName(), introspectionResponseInterceptor,
+                    null);
+            bundleContext.registerService(IntrospectionDataProvider.class.getName(), isIntrospectionDataProvider,
+                    null);
             if (log.isDebugEnabled()) {
-                log.debug("Mutual TLS bundle is activated");
+                log.debug("Mutual TLS bundles are activated");
```

Comment:
no need to change this since all the new classes are packed inside the same bundle

Reply:
Fixed c65eaeb